### DISCUSSION TITLE
Error 500 > SMTP Connect() failed

### DISCUSSION THREAD

**Discussion by bobmoreno** [PHPMailer/PHPMailer#2327] [Q&A]
Hi, dears.
This is my first time with PHPMailer and I have a problem.

In the localhost (EasyPHP), ti's running well. But when I upload it to a host, I receive the message "**Error 500**" or, by the debug line, "**SMTP ERROR: Failed to connect to server: Network is unreachable (101) - SMTP Connect() failed**". 

I'm trying in two hosts, using PHPMailer 6.4.0. Already tried with older versions.

I tried a lot of previous suggestions in this forum.

My Google account is less secure apps enabled.

This password is for test only and may stay public for today.

Someone can help me, please? 

Thank you!

You can access the test page at https://monono.com.br/email/

The PHP script:

<pre>
date_default_timezone_set('America/Sao_Paulo');

// Receive data from form
$nome 		= $_POST["nome"]; // To name
$destino 	= $_POST["destino"]; // To email
$assunto 	= $_POST["assunto"]; // Subject
$mensagem	= $_POST["mensagem"]; // Message
$data		= date("d.m.y"); // Date
$hora		= date("H:i:s"); // Time

// Open PHPMailer
use PHPMailer\PHPMailer\PHPMailer;
use PHPMailer\PHPMailer\Exception;
use PHPMailer\PHPMailer\SMTP;

require 'phpmailer/src/PHPMailer.php';
require 'phpmailer/src/SMTP.php';
require 'phpmailer/src/Exception.php';

// Settings
$mail = new PHPMailer();
$mail->Host = "smtp.gmail.com";
$mail->SetLanguage("pt_br"); 
$mail->IsSMTP(); 
$mail->Host = "smtp.gmail.com";
$mail->SMTPSecure = "tls";
$mail->Port = 587; 
$mail->SMTPAuth = true;
$mail->Username = "monono.contato@gmail.com"; // don't hide, please
$mail->Password = ">lx{t86H"; // don't hide, please
$mail->setFrom('monono.contato@gmail.com', 'Contato Monono');
$mail->AddAddress($destino, $nome);
$mail->IsHTML = true ;
$mail->Subject = $assunto;
$mail->Body = $mensagem. "at ".$data." - ".$hora."</span></p>" ;
$mail->AltBody = $mensagem ;
$mail->SMTPDebug = 2;

if(!$mail->Send())
 { 
 echo "Erro no envio"; // Error message
 echo "Mailer Error: " . $mail->ErrorInfo; 
 } 
 else {
	 echo "Email enviado"; // Success message
 }

</pre>

**Comment by Synchro** [comment]
This is an extremely common problem that is covered extensively in the troubleshooting guide, so please read that. It's most likely that your hosting provider blocks outbound SMTP, which results in the network being unreachable. This has nothing to do with your script or PHPMailer. You will either need to ask them to remove the block, use a different sending mechanism (such as by sending through their own mail servers), or find a better hosting provider.

If that was your real password in your script (I edited it out), I suggest you change it since you've just let the whole world see it.

**Reply by bobmoreno** [reply]
Thank you very much, Synchro. I have tried a lot of previous suggestions in this forum and in the troubleshoting page.  The PHP page was uploaded to a two different hosts - one in Brazil, another in US. I asked him about the bug and they say problem is in the code, because nothing is blocked. This a real Gmail account and real password created just for your tests for today. Please, remain it visible for someone try something. Regards!

**Reply by Synchro** [reply]
Don't believe them! Test it. For example, on a command line on your server run:

    echo QUIT | openssl s_client -starttls smtp -connect smtp.gmail.com:587

If there is no block, the output from that should end with:

    250 SMTPUTF8

If it doesn't, or times out, it's blocked.

**Reply by bobmoreno** [reply]
Thank you! I did it and receive this answer: 
**socket: Bad file descriptor
connect:errno=9**

I just sent it to host support.

**Reply by bobmoreno** [reply]
Believe it or not, support suggest me to search how to solve it in web foruns... :(

**Reply by Synchro** [reply]
Well, if the openssl command line client can't connect, then PHP won't work either, so it's a system or network-wide issue and nothing to do with your script. Unless they provide some other way of sending email from their network, I think you're stuck.

**Comment by CaptainEboy** [comment]
bobmonero I also had the same problem and got it fixed by changing the $mail->Port = 587;

I mean I changed the port number to the servers own.

In your case change 587 to your server's SMTP port number.

That should solve the problem.

Please note that I didn't test your code

**Reply by Synchro** [reply]
Don't use an IMAP port, it won't work. You only need SMTP for outbound, and that will be on ports 25, 465 or 587, depending on your encryption config.

**Reply by CaptainEboy** [reply]
I've edited it, I meant SMTP not IMAP
